src/diffusers/loaders/ip_adapter.py

```diff
@@ -226,7 +226,15 @@ def load_ip_adapter(
 
         # load ip-adapter into unet
         unet = getattr(self, self.unet_name) if not hasattr(self, "unet") else self.unet
-        unet._load_ip_adapter_weights(state_dicts, low_cpu_mem_usage=low_cpu_mem_usage)
+        extra_loras = unet._load_ip_adapter_weights(state_dicts, low_cpu_mem_usage=low_cpu_mem_usage)
```

Comment:
Interesting. To reduce the maintenance burden and to promote better readability, perhaps we could separate out the LoRA-related code from `_load_ip_adapter_weights()`?